inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_b_fs_yx_fsv16_imad.cl

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "include/include_all.cl"
+#include "include/data_types.cl"
+
+#define ALIGN_TO(val, multiple) (((val) + (multiple) - 1) / (multiple) * (multiple))
+
+#define AS_TYPE(type, val) CAT(as_, type)(val)
+#define IN_VEC16 MAKE_VECTOR_TYPE(INPUT0_TYPE, 16)
+#define OUT_VEC16 MAKE_VECTOR_TYPE(OUTPUT_TYPE, 16)
+#define CONVERT_OUT CAT(convert_, OUTPUT_TYPE)
+#define CONVERT_OUT_VEC16 CAT(convert_, OUT_VEC16)
+#define FEATURE_SLICE_SIZE 16
+
+#if MAX_POOLING
+    #define INIT_VAL CHAR_MIN
+#elif AVG_POOLING
+    #define INIT_VAL 0
+#else
+#error
+#endif
+
+
+inline int FUNC(apply_pooling)(int tmp, int in)
+{
+#if MAX_POOLING
+    return max(tmp, in);
+#elif AVG_POOLING
+    return tmp + in;
+#endif
+}
+
+KERNEL(pooling_gpu_b_fs_yx_fsv16)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+)
+{
+    const uint x    = (uint)get_global_id(0);
+    const uint y    = (uint)get_global_id(1);
+    const uint bf   = (uint)get_global_id(2);
+    const uint f    = (bf * FEATURE_SLICE_SIZE) % ALIGN_TO(INPUT0_FEATURE_NUM, FEATURE_SLICE_SIZE);
+    const uint b    = (bf * FEATURE_SLICE_SIZE) / ALIGN_TO(INPUT0_FEATURE_NUM, FEATURE_SLICE_SIZE);
+
+    const int offset_x = (int)x*STRIDE_SIZE_X - PADDING_SIZE_X;
+    const int offset_y = (int)y*STRIDE_SIZE_Y - PADDING_SIZE_Y;
+
+    int result[FEATURE_SLICE_SIZE] = { INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL, INIT_VAL };
+
+#ifdef CHECK_BOUNDRY
```

Comment:
May be CHECK_BOUNDARY is more correct choice?

Reply:
It's something from pooling in general so it would required changes in multiple kernels and pooling base. I'm not sure if should change it while adding one new kernel.

Reply:
Fixing this typo is out of scope for this PR since it would require modifying all other pooling kernels.